libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -0,0 +1,88 @@
+@file:JvmName("NavigationRouteEx")
+package com.mapbox.navigation.base.route
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.DirectionsWaypoint
+import com.mapbox.api.directions.v5.models.RouteOptions
+
+class NavigationRoute(
+    val directionsResponse: DirectionsResponse,
+    val routeIndex: Int,
+    val routeOptions: RouteOptions
+) {
+    val directionsRoute = directionsResponse.routes()[routeIndex]!!
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is NavigationRoute) return false
```

Comment:
missin `inheritors` I think, should be 
```suggestion
       if (javaClass != other?.javaClass) return false
```

Reply:
Good spot! The class is `final`, it cannot have inheritors, so the equality symmetry would always be safe in this specific case. It's interesting that Android Studio started generating this way of type comparison for me though :thinking: 

That said, just for consistency I'll make it based on class.